plugins/tech-insights-backend/src/service/fact/FactRetrieverEngine.ts

```diff
@@ -108,15 +109,24 @@ export class FactRetrieverEngine {
       this.logger.info(
         `Retrieving facts for fact retriever ${factRetriever.id}`,
       );
-      const facts = await factRetriever.handler({
-        ...this.factRetrieverContext,
-        entityFilter: factRetriever.entityFilter,
-      });
-      if (this.logger.isDebugEnabled()) {
-        this.logger.debug(
-          `Retrieved ${facts.length} facts for fact retriever ${
-            factRetriever.id
-          } in ${duration(startTimestamp)}`,
+
+      let facts: TechInsightFact[] = [];
+      try {
+        facts = await factRetriever.handler({
+          ...this.factRetrieverContext,
+          entityFilter: factRetriever.entityFilter,
+        });
+        if (this.logger.isDebugEnabled()) {
```

Comment:
Think we can remove this line as `this.logger.debug` will only log things if the log level is `debug`

Reply:
I think it's my JVM past leaking through. Micro-optimization to not do an expensive string interpolation if it's not needed. Can indeed be removed.